.github/workflows/build-and-publish-docs.yml

```diff
@@ -0,0 +1,31 @@
+name: build-and-publish-docs
+on:
+  push:
+    branches:
+      - master
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container:
+      image: pego/dowhy-examples-notebooks-deps
+    steps:
+      - uses: actions/setup-python@v2
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 0 # otherwise, you will failed to push refs to dest repo
+      - name: Install dependencies
+        run: |
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+          if [ -f requirements-plotting.txt ]; then pip install -r requirements-plotting.txt; fi
+          pip3 install 'scikit-learn==1.0.2' --force-reinstall
+          if [ -f requirements-test.txt ]; then pip install -r requirements-test.txt --use-deprecated=legacy-resolver; fi
+          pip install 'numpy<=1.21' --force-reinstall
+      - name: Build
+        run: |
+          git config --global --add safe.directory /__w/dowhy/dowhy
+          ./docs/generate_docs.sh
+      - name: Commit and push to gh-pages branch
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./dowhy-docs
```

Comment:
currently, the docs need to be updated in a separate branch of dowhy called `gh-pages`. Would this automatically update that branch (without affecting master)?

Reply:
Yes, it updates `gh-pages` without touching `master`/`main`.